backend/appInfo.ts

```diff
@@ -117,26 +114,24 @@ function getManifestInfoFromDir(
 ): ManifestInfo {
   const tomlBuffer = readFileBuffer(path.join(dir, "manifest.toml"));
   if (tomlBuffer === null) {
-    return {
-      name: fallbackName,
-      sourceCodeUrl: undefined,
-      manifestFound: false,
-    };
+    console.error("Missing manifest.toml (from DIR)");
+    return { ...MISSING_MANIFEST, name: fallbackName };
   }
-  const parsed = tomlParse(tomlBuffer.toString());
-  const name = parsed.name || fallbackName;
-  return {
-    name,
-    sourceCodeUrl: parsed.source_code_url,
-    manifestFound: true,
-  };
+  return tomlParse(tomlBuffer.toString(), fallbackName);
 }
 
-function tomlParse(s: string): any {
+function tomlParse(s: string, fallbackName: string): any {
   try {
-    return toml.parse(s);
+    const parsed = toml.parse(s);
+    return {
+      name:
+        parsed.name || fallbackName || "Missing name entry in manifest.toml",
+      sourceCodeUrl: parsed.source_code_url || undefined,
+      manifestFound: true,
+    };
   } catch (e) {
-    throw new AppInfoError("Invalid manifest.toml, please check the format");
+    console.error("Failed to parse manifest.toml, please check the format!");
+    return { ...MISSING_MANIFEST, name: fallbackName };
```

Comment:
If we fail to parse, return `MISSING_MANIFEST` instead of throwing an exception.